src/ResourceManager/Profile/Commands.Profile/AutoSave/EnableAzureRmContextAutosave.cs

```diff
@@ -123,18 +124,19 @@ void EnableAutosave(IAzureSession session, bool writeAutoSaveFile, out ContextAu
             }
         }
 
-        bool IsValidPath(string path)
+        private static bool IsValidPath(string filepath)
         {
             FileInfo valid = null;
             try
             {
-                valid = new FileInfo(path);
+                valid = new FileInfo(filepath);
             }
             catch
             {
                 // swallow any exception
             }
-            return valid != null;
+
+            return valid != null && Directory.Exists(valid.DirectoryName);
```

Comment:
Should use FileUtilities to check existence here as we use it to ensure the directory is created above.